gef.py

```diff
@@ -3146,9 +3188,13 @@ def set_arch(arch=None, default=None):
         except KeyError:
             raise OSError("Specified arch {:s} is not supported".format(arch.upper()))
 
-    current_elf = current_elf or get_elf_headers()
+    if not current_elf:
```

Comment:
We basically assume that a loaded file is an elf. Could we remove `current_elf` in favour of something more general?

Reply:
So i've updated most of the places that use `current_elf` and `get_elf_headers` but there are still `is_arch` which I'm not sure what to do about as I don't know what the `get_arch()` string is for all the types that it's called with. Is there an easy way to find out?